src/tyr/trace_serializer.cc

```diff
@@ -111,21 +111,32 @@ json::ArrayPtr serialize_edges(const AttributesController& controller,
         edge_map->emplace("lane_connectivity", lane_connectivity);
       }
       if (controller(kEdgeMaxDownwardGrade)) {
-        edge_map->emplace("max_downward_grade", static_cast<int64_t>(edge.max_downward_grade()));
+        if (edge.max_downward_grade() == kNoElevationData) {
+          edge_map->emplace("max_downward_grade", static_cast<int64_t>(edge.max_downward_grade()));
+        } else {
+          edge_map->emplace("max_downward_grade", nullptr);
+        }
```

Comment:
the grades also get a `kNoElevationData` in the `TripLegBuilder`

Reply:
The grades aren't an elevation though, they are a slope. Are you saying we store that in the tiles now, I thought they had a different number of buts anyway

Reply:
looks like those are stored in the tiles but not as `kNoElevationData`, that value is only reserved to be stored for elevation: https://github.com/valhalla/valhalla/blob/b889f94452ad9de4661a902cca2ef5f35b2b9229/src/mjolnir/elevationbuilder.cc#L120-L125

the triplegbuilder however already assigned `kNoElevationData` to the slopes when there's no elevation available in the graph tile: https://github.com/valhalla/valhalla/blob/master/src/thor/triplegbuilder.cc#L1092-L1105

happy to change to smth more sensible, for now I just made sure it's also properly used in the serializer and not returned as 32768 (or -500 now)

Reply:
oh i see, thanks for pointing that out! for whatever reason i had thought that we were talking about directed edges here but actually youre talking about the serialization downstream from here.